### ISSUE TITLE
OSRM's edge-expansion [document]

### ISSUE THREAD

**Issue by kkdd** [Project-OSRM/osrm-backend#4851]
Hello,
Is [OSRM's edge-expansion](https://github.com/Project-OSRM/osrm-backend/wiki/Graph-representation) related to the [edge-expansion](https://en.wikipedia.org/wiki/Expander_graph) in graph theory?

**Comment by TheMarex** [comment]
@kkdd No it refers to creating a turn-based graph where edges are turns and nodes are street segments. We need this to model turn restrictions.

**Comment by daniel-j-h** [comment]
@danpat did a beautiful job of visualizing these concepts in:

https://github.com/Project-OSRM/osrm-backend/wiki/Graph-representation

![](https://cloud.githubusercontent.com/assets/1892250/13710991/1b507fb6-e771-11e5-8710-8d5f16d805ce.gif)

![](https://cloud.githubusercontent.com/assets/1892250/19296198/1de51b4a-8ff7-11e6-8339-31eb6361338a.png)

Closing here as not actionable.

**Comment by kkdd** [comment]
Thank you.
What is this transformation called in the literature of graph theory?

**Comment by chaupow** [comment]
Hi @kkdd, I think if you ask about pure graph theory, it's difficult to say because it only applies to graphs with turn restrictions. I guess the original graph is a graph with turn restriction and the edge-expanded graph is a *dual graph* representation.

However, I did a quick google search, and in route planning there seem to be different expressions used:
- [Edge-based graph](http://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.661.3226&rep=rep1&type=pdf)
- [Line graphs](http://geo.fsv.cvut.cz/gdata/2013/pin2/d/dokumentace/line_graph_teorie.pdf)
- [Linear dual graph](https://pdfs.semanticscholar.org/a735/e1cd1b724932b1844d5b21038d287967a59f.pdf)
- [Dual transformation](https://www.hindawi.com/journals/mpe/2017/5839192/)

### tl;dr
**There is no clearly defined name for this transformation**

**Comment by kkdd** [comment]
Hello,
I eventually succeeded to find its name: "[Line graph](https://en.wikipedia.org/wiki/Line_graph#Example)" in Wikipedia!
